tm1638/tm1638.go

```diff
@@ -0,0 +1,137 @@
+/*
+TM1638 is an LED Controller
+*/
+package tm1638
+
+import (
+	"machine"
+	"time"
+)
+
+// Device wraps the pins of the TM1638.
+type Device struct {
+	/* STB - When this pin is "LO" chip accepting transmission */
+	strobe machine.Pin
+	/* CLK - DIO pin reads data at the rising edge and outputs at the falling edge */
+	clock machine.Pin
+	/* DIO - This pin outputs/inputs serial data */
+	data machine.Pin
+}
+
+const (
+	addressAutoIncrement = 0x40
+	fixedAddress         = 0x44
+	baseAddress          = 0xC0
+	maxAddress           = 0x0F
+	readKeys             = 0x42
+	brightness           = 0x88
+	maxBrightness        = 0x07
+	zeroBrightness       = 0x80
+)
+
+// Create new TM1638 device
+func New(strobe machine.Pin, clock machine.Pin, data machine.Pin) Device {
+	strobe.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	clock.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	return Device{strobe: strobe, clock: clock, data: data}
+}
+
+// Configure TM1638
+func (d *Device) Configure() {
+	d.Clear()
+	d.Brightness(maxBrightness)
+}
+
+// Clear display memory
+func (d *Device) Clear() {
+	d.sendCommand(addressAutoIncrement)
+	d.data.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.strobe.Low()
+	d.transmissionDelay()
+	d.write(baseAddress)
+	for i := 0; i < 16; i++ {
+		d.write(0)
+	}
+	d.strobe.High()
+}
+
+// Set display brightness
+func (d *Device) Brightness(value uint8) {
+	if value == 0 {
+		d.sendCommand(zeroBrightness)
+	} else {
+		d.sendCommand(brightness | (value & maxBrightness))
```

Comment:
This can be confusing to users: if they use the value 8 then they suddenly get a dark screen. Instead, I suggest:

* Exporting maxBrightness as `MaxBrightness`
* using `if value > MaxBrightness { value = MaxBrightness }` to limit the value

Reply:
Ok